server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/ShardAssignment.java

```diff
@@ -30,7 +29,10 @@ public int assigned() {
 
     public static ShardAssignment ofAssignedShards(List<ShardRouting> routings) {
         assert routings.stream().allMatch(ShardRouting::started) : routings;
-        var nodeIds = routings.stream().map(ShardRouting::currentNodeId).collect(toCollection(LinkedHashSet::new));
+        var nodeIds = new LinkedHashSet<String>();
```

Comment:
Consider presing `LinkedHashSet` with `Sets.newLinkedHashSetWithExpectedSize`

Reply:
We don't know the size here. It's not a problem though, at least in the profiling I have resize doesn't show up here :)